### PR TITLE
DocForm remove obj

### PR DESCRIPTION
**Pull request by msladek**
- added functionality to remove obj via DocFormCommand (with preceding "^" to objNb)
- refactored request key parsing for DocFormCommand to own parser with return class

### CODE REVIEW

File: src/main/java/com/celements/docform/DocFormRequestKey.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    return objNb;
+  }
+
+  public String getFieldName() {
+    return fieldName;
+  }
+
+  public boolean sameObject(DocFormRequestKey other) {
+    System.out.println("compare " + this);
+    System.out.println("with " + other);
```
Comment (ebeutler) [inline src/main/java/com/celements/docform/DocFormRequestKey.java]: Please change to log or remove.

File: src/test/java/com/celements/web/plugin/cmd/DocFormCommandTest.java [hunk 2]
```diff
@@ -122,67 +87,123 @@ public void testGetUpdateDoc() throws XWikiException {
...
-    verifyAll(obj);
+    
+    replayDefault();
+    docFormCmd.setOrRemoveObj(docMock, key, value, context);
+    verifyDefault();
+    
+    assertEquals(objMock, docFormCmd.getChangedObjects().get("W:Sp.Doc_W:A.B_0"));
   }
   
+  // TODO do nothing in this case?
```
Comment (ebeutler) [inline src/test/java/com/celements/web/plugin/cmd/DocFormCommandTest.java]: Still needed?

**Comment by ebeutler** [comment]
Two small things, rest looks fine.

**Comment by fpichler** [comment]
Please do NOT remove any public methods from ScriptService classes (e.g. CelementsWebScriptService) instead mark them deprecated and add a delegation to the new ScriptService method. Otherwise all velocity scripts using this public methods will brake!

**Comment by fpichler** [comment]
Please do NOT use commons-logging any more. Thus replace all commons-LOGGER in new classes with slf4j logger.

**Comment by fpichler** [comment]
sorry, wrong pull request for the last to messages. :-(

**Pull request merged**